pkg/signature/ed25519.go

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package signature
+
+import (
+	"context"
+	"crypto"
+	"crypto/ed25519"
+	"crypto/rand"
+	_ "crypto/sha256" // To ensure `crypto.SHA256` is implemented.
+	"errors"
+	"fmt"
+)
+
+type ED25519Verifier struct {
+	Key ed25519.PublicKey
+}
+
+type ED25519SignerVerifier struct {
+	ED25519Verifier
+	Key ed25519.PrivateKey
+}
+
+func (s ED25519SignerVerifier) Sign(_ context.Context, rawPayload []byte) (signature, signed []byte, err error) {
+	signature = ed25519.Sign(s.Key, rawPayload)
+	return
+}
```

Comment:
might be worth capturing an err here, such as:

https://github.com/sigstore/sigstore/blob/11e6eaab7cdc5cea7113004d2de344fe0f7fb640/pkg/signature/ecdsa.go#L45-L48

Reply:
The [ed25519.Sign](https://golang.org/pkg/crypto/ed25519/#Sign) function actually only returns `[]byte` :)